smart_card_connector_app/src/application_unittest.cc

```diff
@@ -282,14 +290,24 @@ class SmartCardConnectorApplicationTest : public ::testing::Test {
                        /*function_name=*/"SCardReleaseContext",
                        ArrayValueBuilder().Add(scard_context).Get());
     waiter->Wait();
-    Value reply = std::move(*waiter).take_value();
-
-    GOOGLE_SMART_CARD_CHECK(reply.is_array());
-    GOOGLE_SMART_CARD_CHECK(reply.GetArray().size() == 1);
+    return ExtractReturnCodeAndResults(std::move(*waiter).take_value());
+  }
 
-    GOOGLE_SMART_CARD_CHECK(reply.GetArray()[0]->is_integer());
-    LONG return_code = reply.GetArray()[0]->GetInteger();
-    return return_code;
+  LONG SimulateListReadersJsCall(int handler_id,
+                                 SCARDCONTEXT scard_context,
+                                 Value groups,
+                                 std::vector<std::string>& out_readers) {
+    const RequestId request_id = ++request_id_counter_;
+    const std::string requester_name = GetJsClientRequesterName(handler_id);
+    auto waiter =
+        global_context_.CreateResponseWaiter(requester_name, request_id);
+    SimulateCallFromJs(
+        requester_name, request_id,
+        /*function_name=*/"SCardListReaders",
+        ArrayValueBuilder().Add(scard_context).Add(std::move(groups)).Get());
+    waiter->Wait();
```

Comment:
Can this also be refactored into a method with inputs handler_id, function_name and an ArrayValue for the args?

Reply:
Good idea, done.